### PR TITLE
Make identityRef required on AzureManagedControlPlane

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**: IdentityRef is required since https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3793, fixing the CRD to reflect that.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

- [x] cherry-pick candidate

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
 identityRef is required on AzureManagedControlPlane Spec
```

### CODE REVIEW

**Comment by CecileRobertMichon** [comment]
/cherry-pick release-1.11

not need to cherry-pick to 1.10 since https://github.com/kubernetes-sigs/cluster-api-provider-azure/commit/1a94893c23bd8558cbe0bff2d9e3b975d58c812b is only in 1.11

**Review by nojnhuh — COMMENTED** [review]
/lgtm
/approve

File: api/v1beta1/azuremanagedcontrolplane_types.go [hunk 1]
```diff
@@ -160,8 +160,7 @@ type AzureManagedControlPlaneSpec struct {
 	LoadBalancerSKU *string `json:"loadBalancerSKU,omitempty"`
 
 	// IdentityRef is a reference to a AzureClusterIdentity to be used when reconciling this cluster
-	// +optional
-	IdentityRef *corev1.ObjectReference `json:"identityRef,omitempty"`
+	IdentityRef *corev1.ObjectReference `json:"identityRef"`
```
Comment (nojnhuh) [inline api/v1beta1/azuremanagedcontrolplane_types.go:163]: /hold
Should we make the same change for AzureCluster?

Comment (CecileRobertMichon) [reply api/v1beta1/azuremanagedcontrolplane_types.go:163]: For AzureCluster the type is on the clusterclass and it's already enforced at a per cluster level with https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/api/v1beta1/azurecluster_validation.go#L145

Comment (nojnhuh) [reply api/v1beta1/azuremanagedcontrolplane_types.go:163]: Makes sense!
/hold cancel

Comment (CecileRobertMichon) [reply api/v1beta1/azuremanagedcontrolplane_types.go:163]: /hold

Would it be better to make it required everywhere and add the webhook validation to AzureManagedControlPlane as well for consistency?

**Comment by CecileRobertMichon** [comment]
@nojnhuh I pushed a second commit that makes the validation consistent across AzureCluster and AzureManagedControlPlane, let me know if you think that's better. If not I can revert the last commit.

File: api/v1beta1/azuremanagedcontrolplane_webhook.go [hunk 2]
```diff
@@ -872,3 +873,14 @@ func (m *AzureManagedControlPlane) validateNetworkPluginMode(_ client.Client) er
 
 	return nil
 }
+
+// validateIdentityRef validates an IdentityRef.
+func (m *AzureManagedControlPlane) validateIdentityRef(_ client.Client) error {
```
[the remark this answers is not in the document]
Comment (CecileRobertMichon) [reply api/v1beta1/azuremanagedcontrolplane_webhook.go:878]: given this is a bug fix we want to cherry-pick, let's go for least risky / least amount of change possible for now. I will revert the commit, and let @willie-yao refactor the validation along with the others in #4155.

**Comment by CecileRobertMichon** [comment]
reverted the last commit

/hold cancel

**Pull request merged**